src/attribute-editor/row.tsx

```diff
@@ -14,39 +14,34 @@ import clsx from 'clsx';
 
 const Divider = () => <InternalBox className={styles.divider} padding={{ top: 'l' }} />;
 
-interface AttributeProps {
-  info?: React.ReactNode;
-  label?: React.ReactNode;
-  errorText?: React.ReactNode;
-  control?: React.ReactNode;
-  constraintText?: React.ReactNode;
-  hideLabel?: boolean;
-}
-
-const Attribute = ({ control, hideLabel, ...props }: AttributeProps) => (
-  <InternalFormField __hideLabel={hideLabel} {...props} className={styles.field} stretch={true}>
-    {control}
-  </InternalFormField>
-);
-
 export interface RowProps<T> {
   breakpoint: ColumnLayoutBreakpoint | null;
   item: T;
   definition: ReadonlyArray<AttributeEditorProps.FieldDefinition<T>>;
+  i18nStrings: AttributeEditorProps.I18nStrings | undefined;
   index: number;
   removable: boolean;
   removeButtonText: string;
   removeButtonRefs: Array<ButtonProps.Ref | undefined>;
   onRemoveButtonClick?: NonCancelableEventHandler<AttributeEditorProps.RemoveButtonClickDetail>;
 }
 
+function render<T>(
+  item: T,
+  itemIndex: number,
+  slot: AttributeEditorProps.FieldRenderable<T> | React.ReactNode | undefined
```

Comment:
Is the `undefined` needed?

Reply:
It is implicit inside `ReactNode`, but better to put it here clearly to show that slot may be missing